src/nnp_force.F

```diff
@@ -0,0 +1,682 @@
+         unit_nr = cp_print_key_unit_nr(logger, print_key, extension=".xyz", &
+                                        middle_name="nnp-bias-forces")
+         CALL nnp_print_bias_forces(nnp, unit_nr)
+         CALL cp_print_key_finished_output(unit_nr, logger, print_key)
+      END IF
+
+   END SUBROUTINE nnp_print_bias
+
+! **************************************************************************************************
+!> \brief Print NNP bias energies
+!> \param nnp ...
+!> \param unit_nr ...
+!> \param file_is_new ...
+!> \date   2020-10-10
+!> \author Christoph Schran (christoph.schran@rub.de)
+! **************************************************************************************************
+   SUBROUTINE nnp_print_bias_energy(nnp, unit_nr, file_is_new)
+      TYPE(nnp_type), POINTER                            :: nnp
+      INTEGER, INTENT(IN)                                :: unit_nr
+      LOGICAL, INTENT(IN)                                :: file_is_new
+
+      CHARACTER(len=*), PARAMETER :: routineN = 'nnp_print_bias_energy', &
+         routineP = moduleN//':'//routineN
+
+      CHARACTER(len=default_path_length)                 :: fmt_string
+      INTEGER                                            :: i
+
+      IF (unit_nr > 0) THEN
+         IF (file_is_new) THEN
+            WRITE (unit_nr, "(A1)", ADVANCE='no') "#"
+            WRITE (unit_nr, "(2(2X,A19))", ADVANCE='no') "Sigma [a.u.]", "Bias energy [a.u.]"
+            DO i = 1, nnp%n_committee
+               IF (nnp%bias_align) THEN
+                  WRITE (unit_nr, "(2X,A16,I3)", ADVANCE='no') "shifted E_NNP", i
+               ELSE
+                  WRITE (unit_nr, "(2X,A16,I3)", ADVANCE='no') "E_NNP", i
+               END IF
+            END DO
+            WRITE (unit_nr, "(A)") ""
+
+         END IF
+
+         WRITE (fmt_string, "(A,I3,A)") "(2X,", nnp%n_committee + 2, "(F20.9,1X))"
+         WRITE (unit_nr, fmt_string) nnp%bias_sigma, nnp%bias_energy, nnp%committee_energy
+      END IF
+
+   END SUBROUTINE nnp_print_bias_energy
+
+! **************************************************************************************************
+!> \brief Print NNP bias forces
+!> \param nnp ...
+!> \param unit_nr ...
+!> \date   2020-10-10
+!> \author Christoph Schran (christoph.schran@rub.de)
+! **************************************************************************************************
+   SUBROUTINE nnp_print_bias_forces(nnp, unit_nr)
+      TYPE(nnp_type), POINTER                            :: nnp
+      INTEGER, INTENT(IN)                                :: unit_nr
+
+      CHARACTER(len=*), PARAMETER :: routineN = 'nnp_print_bias_forces', &
+         routineP = moduleN//':'//routineN
+
+      CHARACTER(len=default_path_length)                 :: fmt_string
+      INTEGER                                            :: i
+
+      ! TODO use write_particle_coordinates from particle_methods.F
+      IF (unit_nr > 0) THEN
```

Comment:
since this is the only condition under which this function prints something and there's a single caller as far as I've seen, you could do `IF (unit_nr > 0) CALL nnp_print_bias_force(nnp)` instead?
Same goes for `nnp_print_sumforces`?

Reply:
That's actually valid for almost all the print functions. Will change that, thanks!